packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -298,27 +364,44 @@ export function resetHooks(): void {
 
   if (__DEV__) {
     currentHookNameInDev = undefined;
+    verifyHookCallOrder();
   }
+  currentHookType = null;
+  currentHookMatches = [];
 
   didScheduleRenderPhaseUpdate = false;
   renderPhaseUpdates = null;
   numberOfReRenders = -1;
 }
 
 function createHook(): Hook {
-  return {
-    memoizedState: null,
+  let hook: Hook = __DEV__
+    ? {
+        // $FlowFixMe - we're certain we've set currentHookType by now
```

Comment:
We use FlowFixMe for bugs in Flow — I think this just needs to be conversion via any. `(foo: any): desiredtype`